overlord/snapstate/snapstate_test.go

```diff
@@ -176,9 +185,11 @@ func verifyInstallUpdateTasks(c *C, opts, discards int, ts *state.TaskSet, st *s
 	}
 	expected = append(expected,
 		"set-auto-aliases",
-		"setup-aliases",
-		"start-snap-services",
-	)
+		"setup-aliases")
+	if opts&install != 0 {
+		expected = append(expected, "run-hook")
```

Comment:
Ok, I added checks to the generic TestInstallTasks / TestRemoveTasks and made sure these are the hooks we expect, plus added two new tests to make sure these hooks are not run when they shouldn't.